components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -142,20 +123,12 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 			projectsDataByte:    projectsDataByte,
 		}
 
-		if len(node.GetSourceId()) == 0 || len(node.GetSourceAccountId()) == 0 || len(node.GetSourceRegion()) == 0 {
+		if !hasCloudInformation(node) {
```

Comment:
This just reads easier for me.

Reply:
and documents/reveals the design intent 💯